kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/ExpositionIT.java

```diff
@@ -143,6 +144,60 @@ void exposesTwoClusterOverPlainWithSeparatePorts(KafkaCluster cluster) {
+     * possible for Kroxylicious to change the port it advertises its brokers at. So that clients will be told to connect
+     * to the advertised port (e.g. 443) rather than the listening port for the VirtualCluster.
+     */
+    @Test
+    void exposesUpstreamClustersUsingSniRoutingBehindPassthroughProxy(KafkaCluster cluster) throws Exception {
+        try (var proxy = new PassthroughProxy(9192, "localhost")) {
+            var virtualClusterCommonNamePattern = IntegrationTestInetAddressResolverProvider.generateFullyQualifiedDomainName(".cluster");
+            var virtualClusterBootstrapPattern = "bootstrap" + virtualClusterCommonNamePattern;
+            var virtualClusterBrokerAddressPattern = "broker-$(nodeId)" + virtualClusterCommonNamePattern;
+
+            var builder = new ConfigurationBuilder();
+
+            var base = new VirtualClusterBuilder()
+                    .withNewTargetCluster()
+                    .withBootstrapServers(cluster.getBootstrapServers())
+                    .endTargetCluster()
+                    .build();
+
+            var keystoreTrustStorePair = buildKeystoreTrustStorePair("*" + virtualClusterCommonNamePattern);
+
+            var virtualCluster = new VirtualClusterBuilder(base)
+                    .withClusterNetworkAddressConfigProvider(
+                            new ClusterNetworkAddressConfigProviderDefinitionBuilder(SniRoutingClusterNetworkAddressConfigProvider.class.getName())
+                                    .withConfig("bootstrapAddress", virtualClusterBootstrapPattern + ":9192",
+                                            "advertisedBrokerAddressPattern", virtualClusterBrokerAddressPattern + ":" + proxy.getLocalPort())
+                                    .build())
+                    .withNewTls()
+                    .withNewKeyStoreKey()
+                    .withStoreFile(keystoreTrustStorePair.brokerKeyStore())
+                    .withNewInlinePasswordStoreProvider(keystoreTrustStorePair.password())
+                    .endKeyStoreKey()
+                    .endTls()
+                    .withLogNetwork(true)
+                    .withLogFrames(true)
+                    .build();
+            builder.addToVirtualClusters("cluster", virtualCluster);
+
+            try (var tester = kroxyliciousTester(builder)) {
+                // the tester is aware that it should connect to the Virtual Cluster's advertised port
+                try (var admin = tester.admin("cluster", Map.of(
+                        CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SSL.name,
+                        SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, keystoreTrustStorePair.clientTrustStore(),
+                        SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, keystoreTrustStorePair.password()))) {
+                    // do some work to ensure virtual cluster is operational
+                    createTopic(admin, TOPIC, 1);
```

Comment:
I think it is worth describing the cluster and verifying that the ports are truely proxy.getLocalPort().

Reply:
Yep, at one point this test was passing, but the brokers were still being exposed as 9092. Will add

Reply:
added the check:

```
admin.describeCluster().nodes().get(5, TimeUnit.SECONDS).forEach(node -> {
                        assertThat(node.port()).isEqualTo(proxy.getLocalPort());
});
```